### PR TITLE
Add support of jetty 12 ee10/ee9

### PR DESCRIPTION
**Pull request by olamy**
#### Short description of what this resolves:

Adding support of Jetty 12

#### Changes proposed in this pull request:

- add support of Jetty 12 ee10
- add support of Jetty 12 ee9

### CODE REVIEW

File: jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/ArquillianAppProvider.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+         * Use of java.io.tmpdir Should be a last-resort fallback for temp directory.
+         * 
+         * Use of java.io.tmpdir on CI systems is dangerous (overwrite possibility is extremely high)
+         * 
+         * Use of java.io.tmpdir on Unix systems is unreliable (due to common /tmp dir cleanup processes)
+         */
+        File systemDefaultTmpDir = new File(AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("java.io.tmpdir")));
+
+        // If running under maven + surefire, use information provided by surefire.
+        String baseDirVal = AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("basedir"));
```
Comment (joakime) [inline jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/ArquillianAppProvider.java:56]: oh no!

AccessController is deprecated in the Java 17 btw.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/AccessController.html

File: jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedContainer.java [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.jboss.arquillian.spi.client.container.DeployableContainer#deploy(org.jboss.shrinkwrap.descriptor.api.Descriptor)
+     */
+    public void deploy(Descriptor descriptor) throws DeploymentException {
+        throw new UnsupportedOperationException("Not implemented");
```
Comment (joakime) [inline jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedContainer.java:235]: The DeploymentManager should be able to handle this

File: jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedContainer.java [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.jboss.arquillian.spi.client.container.DeployableContainer#undeploy(org.jboss.shrinkwrap.descriptor.api.Descriptor)
+     */
+    public void undeploy(Descriptor descriptor) throws DeploymentException {
+        throw new UnsupportedOperationException("Not implemented");
```
Comment (joakime) [inline jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/JettyEmbeddedContainer.java:244]: The DeploymentManager should be able to handle this

File: jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/ArquillianAppProvider.java [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+         * Use of java.io.tmpdir Should be a last-resort fallback for temp directory.
+         * 
+         * Use of java.io.tmpdir on CI systems is dangerous (overwrite possibility is extremely high)
+         * 
+         * Use of java.io.tmpdir on Unix systems is unreliable (due to common /tmp dir cleanup processes)
+         */
+        File systemDefaultTmpDir = new File(AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("java.io.tmpdir")));
+
+        // If running under maven + surefire, use information provided by surefire.
+        String baseDirVal = AccessController.doPrivileged((PrivilegedAction<String>) () -> System.getProperty("basedir"));
```
Comment (joakime) [inline jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/ArquillianAppProvider.java:55]: Deprecated AccessController again.

File: jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedContainer.java [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.jboss.arquillian.spi.client.container.DeployableContainer#deploy(org.jboss.shrinkwrap.descriptor.api.Descriptor)
+     */
+    public void deploy(Descriptor descriptor) throws DeploymentException {
+        throw new UnsupportedOperationException("Not implemented");
```
Comment (joakime) [inline jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedContainer.java:234]: The DeploymentManager should be able to handle this

File: jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedContainer.java [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.jboss.arquillian.spi.client.container.DeployableContainer#undeploy(org.jboss.shrinkwrap.descriptor.api.Descriptor)
+     */
+    public void undeploy(Descriptor descriptor) throws DeploymentException {
+        throw new UnsupportedOperationException("Not implemented");
```
Comment (joakime) [inline jetty-embedded-12-ee9/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee9/JettyEmbeddedContainer.java:243]: The DeploymentManager should be able to handle this

**Review by joakime — COMMENTED** [review]
Get rid of the AccessController use in Java 17 minimum projects.

File: jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/ArquillianAppProvider.java (hunk 1 above)
Comment (olamy) [reply jetty-embedded-12-ee10/src/main/java/org/jboss/arquillian/container/jetty/embedded_12_ee10/ArquillianAppProvider.java]: good catch done with cbf57cc01b5edd809a9f0dd15075e7ad9a88c9bc

**Comment by olamy** [comment]
@starksm64 could we get a release with this change? thanks!

**Pull request merged**